core/src/avm2/class.rs

```diff
@@ -678,34 +680,74 @@ impl<'gc> Class<'gc> {
                 let mut did_override = false;
 
                 while let Some(superclass) = current_superclass {
-                    let superclass_def = superclass.inner_class_definition();
-
-                    for supertrait in &*superclass_def.traits() {
+                    for supertrait in &*superclass.traits() {
                         let super_name = supertrait.name();
                         let my_name = instance_trait.name();
 
                         let names_match = super_name.local_name() == my_name.local_name()
                             && (super_name.namespace().matches_ns(my_name.namespace())
                                 || (is_protected
-                                    && superclass_def
-                                        .protected_namespace()
-                                        .map_or(false, |prot| {
-                                            prot.exact_version_match(super_name.namespace())
-                                        })));
+                                    && superclass.protected_namespace().map_or(false, |prot| {
+                                        prot.exact_version_match(super_name.namespace())
+                                    })));
                         if names_match {
                             match (supertrait.kind(), instance_trait.kind()) {
                                 //Getter/setter pairs do NOT override one another
                                 (TraitKind::Getter { .. }, TraitKind::Setter { .. }) => continue,
                                 (TraitKind::Setter { .. }, TraitKind::Getter { .. }) => continue,
-                                (_, _) => did_override = true,
-                            }
-
-                            if supertrait.is_final() {
-                                return Err(format!("VerifyError: Trait {} in class {} overrides final trait {} in class {}", instance_trait.name().local_name(), superclass_def.name().local_name(), supertrait.name().local_name(), superclass_def.name().local_name()).into());
-                            }
 
-                            if !instance_trait.is_override() {
-                                return Err(format!("VerifyError: Trait {} in class {} has same name as trait {} in class {}, but does not override it", instance_trait.name().local_name(), self.name().local_name(), supertrait.name().local_name(), superclass_def.name().local_name()).into());
+                                (TraitKind::Const { .. }, TraitKind::Const { .. })
+                                | (TraitKind::Const { .. }, TraitKind::Slot { .. })
+                                | (TraitKind::Slot { .. }, TraitKind::Const { .. })
+                                | (TraitKind::Slot { .. }, TraitKind::Slot { .. }) => {
+                                    did_override = true;
+
+                                    // Const/Var traits override each other in avmplus
```

Comment:
I don't think this is entirely true, as in: FP is even more lax. It lets `var/const/class` (though ofc the last one won't happen in the wild) "override" any trait. Here's my old demo:
```as
class B {
    public var field1 = "base.field1";
    public function get field2(){ return "base.field2"; }
    public function field3(){ return "base.field3"; }
    public function printBase(){
        trace(this.field1); // base.field1
        trace(this.field2); // base.field2
        trace(this.field3); // child.field3
        trace(this["field1"]); // child.field1
        trace(this["field2"]); // child.field2
        trace(this["field3"]); // child.field3
    }
}
class C extends B {
    public var field11 = "child.field1";
    public var field21 = "child.field2";
    public var field31 = "child.field3";
}
new C().printBase();
```
(with `child.field3` showing up because optimizer couldn't optimize getting a bound method, so it just left a generic `getproperty`)